lib/inspec-reporter-json-hdf/reporter.rb

```diff
@@ -0,0 +1,162 @@
+    def render
+      output(report.to_json, false)
+    end
+
+    def self.run_data_schema_constraints
+      '~> 0.0' # Accept any non-breaking change
+    end
+
+    def report
+      report = Inspec::Reporters::Json.new(@config).report
+      attestations = collect_attestations
+
+      report[:profiles].each do |profile|
+        profile[:controls].each do |control|
+          attestation = attestations.detect { |x| x['control_id'].eql?(control[:id]) }
+
+          next if attestation.nil?
+
+          control[:attestation] = attestation
+          unless attestation_expired?(attestation['updated'], attestation['frequency'])
+            control[:results] = apply_attestation(control[:results], attestation)
+          end
+        end
+      end
+      report
+    end
+
+    private
+
+    def apply_attestation(results, attestation)
+      if results.empty?
+        results = [{
+          "code_desc": 'Manually verified Status provided through attestation',
+          "run_time": 0.0,
+          "start_time": DateTime.now.to_s,
+          "status": attestation['status'],
+          "message": attestation_message(attestation)
+        }]
+      else
+        results.each do |result|
+          result[:message] = 'Automated test returned as passed.' if result[:status].eql?('passed')
+          result[:message] = result[:skip_message] if result[:status].eql?('skipped')
+
+          result[:status] = attestation['status']
+          result[:message] = result[:message] + attestation_message(attestation)
+
+          if result[:backtrace]
+            result[:message] = result[:message] + "\nbacktrace: #{result[:backtrace]}"
+            result[:backtrace] = nil
+          end
+        end
+      end
+      results
+    end
+
+    def attestation_message(attestation)
+      "
+       Attestation:
+       Status: #{attestation['status']}
+       Explanation: #{attestation['explanation']}
+       Updated: #{attestation['updated']}
+       Updated By: #{attestation['updated_by']}
+       Frequency: #{attestation['frequency']}
+       "
+    end
+
+    def attestation_expired?(date, frequency)
+      (advanced_date(date, frequency).mjd - DateTime.now.mjd).negative?
+    end
+
+    def advanced_date(date, frequency)
+      parsed_date = DateTime.strptime(date, DATE_FORMAT)
+
+      case frequency.downcase
+      when 'annually'
+        parsed_date.next_year(1)
+      when 'semiannually'
+        parsed_date.next_year(0.5)
+      when 'quarterly'
+        parsed_date.next_year(0.25)
+      when 'monthly'
+        parsed_date.next_month(1)
+      when 'every2weeks'
+        parsed_date.next_day(14)
+      when 'weekly'
+        parsed_date.next_day(7)
+      when 'every3days'
+        parsed_date.next_day(3)
+      when 'daily'
+        parsed_date.next_day(1)
+      else
+        parsed_date
+      end
+    end
+
+    def valid_date?(date)
+      (DateTime.strptime(date, DATE_FORMAT).mjd - DateTime.now.mjd) <= 0
```

Comment:
Does this need to have any logic in it other than an attempt to parse the `DateTime`?

```ruby
return true if DateTime.strptime(date, DATE_FORMAT)
```

Reply:
yes I wanted to fail if the date provided was in the future. Seemed meaningful to make the date invalid in that case.

Reply:
In that case couldn't it just be `DateTime.strptime(date, DATE_FORMAT) > DateTime.now`?